server/src/main/java/com/google/sps/utils/AdRowProcessor.java

```diff
@@ -79,26 +91,112 @@ public static long getImpressionsMin(String impressionsField) throws IllegalArgu
     return 0;
   }
 
+  /*
+   * getImpressionsMax tries to parse the maximum number of impressions and throws an error
+   * if the field is not parsable.
+   */ 
   public static long getImpressionsMax(String impressionsField) throws IllegalArgumentException, NumberFormatException {
     String[] impressionsArray = formatImpressionField(impressionsField); 
     return Long.parseLong(impressionsArray[impressionsArray.length - 1]);   
   }
 
-  public static boolean getAgeTargets(String str) {
-    if (str.trim().equals("Not targeted")) {
+  /* 
+   * Helper function for getImpressionsMin and getImpressionsMax.
+   * Min and max impression fields contain letters k and M as well as ≤. The 
+   * formatImpressionField helper method makes the fields easier to parse later.
+   */ 
+  public static String[] formatImpressionField(String str) {
+    String impressionsString = str.replace("k", "000").replace("M", "000000").replace(" ","").replace("≤", "");
+    String[] impressionsArray = impressionsString.split("-");
+    return impressionsArray; 
+  }
+
+  /* 
+   * isTargetingAge determines whether or not the ad is targeting age.
+   */ 
+  public static boolean isTargetingAge(String str) {
+    if (str.trim().toLowerCase().equals("not targeted")) {
       return false; 
     }
     return true; 
   }
 
+  /* 
+   * checkGenderTarget determines if the arguments in the gender target field are valid.
+   * If they are not valid, the function will throw an exception.
+   */
+  public static List<String> checkGenderTarget(String str) throws IllegalArgumentException {
+    List<String> targets = convertStringToList(str);
+    List<String> validGenderTargets = Arrays.asList("not targeted", "female", "male", "unknown gender");
+    for (String target : targets) {
+      if (!validGenderTargets.contains(target.trim().toLowerCase())) {
```

Comment:
It doesn't really matter in this case b/c `validGenderTargets` is small and not gonna grow, but using a hashset will be faster if all `validGenderTargets` is used for is to check membership.